app/controllers/profiles_controller.rb

```diff
@@ -9,7 +9,15 @@ class ProfilesController < ApplicationController
 
   def show
     @user = User.find_by_slug!(params[:id])
-    @rubygems = @user.rubygems_downloaded.includes(%i[latest_version gem_download]).strict_loading
+    return @rubygems = @user.rubygems_downloaded.includes(%i[latest_version gem_download]).strict_loading if @user.email_confirmed?
+    respond_to do |format|
+      format.any do
+        render plain: t(:this_rubygem_could_not_be_found), status: :not_found
+      end
+      format.html do
+        render file: Rails.public_path.join("404.html"), status: :not_found, layout: false, formats: [:html]
+      end
+    end
```

Comment:
We have a `render_not_found` method that can simplify this:

```suggestion
    return render_not_found unless @user.email_confirmed?
    @rubygems = @user.rubygems_downloaded.includes(%i[latest_version gem_download]).strict_loading
```

Reply:
Done, thanks for the hint!